### PR TITLE
Fix panic in prom_sd_processor when address is empty

### PR DESCRIPTION
**Pull request by mapno**
#### PR Description

Fix panic in prom_sd_processor when address is empty and configs docs

#### Which issue(s) this PR fixes 

Fixes https://github.com/grafana/agent/issues/1276 and https://github.com/grafana/agent/issues/1277

#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ for detailed information.
 - [BUGFIX] Grafana Agent Operator: The /-/ready and /-/healthy endpoints will
   no longer always return 404 (@rfratto).
 
+- [BUGFIX] Fix panic in prom_sd_processor when address is empty (@mapno)
```
[the remark this answers is not in the document]
Comment (mapno) [reply CHANGELOG.md]: Oh, I added it under v0.22.0, my bad! We usually release patch releases or wait for the next minor. cc/ @rfratto

**Review by rfratto — APPROVED** [review]
LGTM. How critical is this bugfix? Is there a known workaround to the panic?

**Comment by mapno** [comment]
> LGTM. How critical is this bugfix? Is there a known workaround to the panic?

Kind of. The workaround is not using the `connection` pod association if you're not sure that it's always set (AFAIK, out of the official OTel receivers, only OTLP implements the necessary code).

**Comment by rfratto** [comment]
Hmm, ok, sounds like this is a good candidate for an emergency patch release, then?

**Pull request merged**